src/platform/silabs/wifi/icd/WifiSleepManager.cpp

```diff
@@ -158,9 +158,11 @@ CHIP_ERROR WifiSleepManager::VerifyAndTransitionToLowPowerMode()
     return ConfigureDTIMBasedSleep();
 
 #elif RS911X_WIFI // rs9116
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
     VerifyOrReturnError(ConfigurePowerSave() == SL_STATUS_OK, CHIP_ERROR_INTERNAL);
+#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
```

Comment:
the #if isn't necessary - WifiSleepManager is only included for icd builds.

Reply:
If we set `CHIP_CONFIIG_ENABLE_ICD_SERVER` to 0 or undefine it, the sleep manager still gets included right? I know the component if not used will not include this.

Reply:
No, the sleep manager is only included for now on ICD builds is gatted by the build flag or the component. This isn't necessary.

Reply:
I am not sure what I am doing wrong, but can you check by setting the following options in `matter_icd_core.slcc`
```yaml
- name: CHIP_CONFIG_ENABLE_ICD_SERVER
  value: 0
- name: SL_ICD_ENABLED
  value: 0
- name: SL_SI91X_NPSS_GPIO_BTN_HANDLER
  value: 0
```

Reply:
Why would we disable the ICD features in the icd_core component? I'm not sure i understand what we are trying to test or achieve here...